### ISSUE TITLE
🚑️ Hide "short" draft trips

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#966]
Report from a program admin:

> This has been the complaint I have gotten from multiple app users since the update:
> What is up with this app? I haven't gone anywhere since I got home and the app is asking me for updates and I've had to do 11 "not a trip" labels.
> The map clearly shows that I haven't moved so I don't know why it's asking me for this.

We haven't changed the native code on iOS at all, and we haven't changed the notification code at all.

For the record, I don’t see a huge increase (or any increase, really) in the number of  “Not a trip” labels in the program. So I think that this is related to the change to show draft trips in the diary. We sometimes get an inadvertent geofence exit from the phone, but once we start tracking, we detect that we didn’t really move very much, so we filter out the trips out when we process the data.

HOWEVER, while the trips are still pending processing, they will show up as draft trips. And if the user is opening their app multiple times a day, they will see the trips as draft, they will mark them as "not a trip" and then we will strip out the trips anyway in post-processing, so we will never see the labels on the public dashboard.

Proposed fix: we can stop showing draft trips that are below a certain threshold (e.g. < 100m in length) to reduce confusion.

@JGreenlee @Abby-Wheelis can we roll this into the next release?

**Comment by shankari** [comment]
I wonder if the"updates" they are talking about are the prompts for #967 
And then they launch the app and just see a bunch of draft trips....

**Comment by JGreenlee** [comment]
100m sounds good, but I suggest that we use the "net distance" - the direct distance between the start place and end place without considering the trajectory.

Most of the things I label "Not a trip" are while I'm at home, it draws a straight line to the end of my block and then a straight line back to my home.

So the distance is 2 blocks but the "net distance" is 0.

**Comment by shankari** [comment]
Alas, that will not work because it will also filter out round trips like this one that I took today.
We can check what we implement on the server and implement the same - I think it is ~ 100m trajectory distance, close to zero O-D distance and less than 5 minutes in length.

<img width="303" alt="Screenshot 2023-09-05 at 8 27 13 PM" src="https://github.com/e-mission/e-mission-docs/assets/2423263/6c564064-0bf1-4fc6-b0ab-7b948e44aaf0">

**Comment by JGreenlee** [comment]
That is not always enough though...
.066mi = 106 meters

![1000002740.png](https://github.com/e-mission/e-mission-docs/assets/15843932/3137f7b7-ee6e-42f1-b19b-d46977174d09)

**Comment by shankari** [comment]
well, it is not going to be perfect, and this trip would have shown up in the label screen before as well. And we didn't get a huge pushback on the pre-draft trip days. Let me look up what the filtering criteria on the server actually is.

**Comment by shankari** [comment]
Most of the filtering on the server seems to happen in `emission/analysis/intake/cleaning/clean_and_resample.py`
and is all focused on 100 meters. I would be OK to add both 100 meters *and* 5 minutes.
I could swear that is what I did but I can't find the code for it right now.

I can ask for the list of users that complained and we can see how many of their draft trips would be fixed by this hueristic

```
 218     trip_distance = sum([section.data.distance for section in list(section_map.values())]) + \
 219                     sum([stop.data.distance for stop in list(stop_map.values())])
 220     filtered_trip_data.distance = trip_distance
 221     filtered_trip_entry["data"] = filtered_trip_data
 222     if filtered_trip_data.distance < 100:
 223         logging.info("Skipped single point trip %s (%s -> %s) of length %s" %
 224                      (trip.get_id(), trip.data.start_fmt_time,
 225                       trip.data.end_fmt_time, filtered_trip_data.distance))
```

```
1063     if distance_delta > 100:
1064         logging.debug("distance_delta = %s > 100, abandoning squish" %
1065                       (distance_delta))
1066         return
1067     else:
1068         logging.debug("distance_delta = %s < 100, continuing with squish " %
1069                       (distance_delta))
```

**Comment by JGreenlee** [comment]
Filtering out <100m and <5min was implemented in [24c5856bd3066bf6e9ac367caabd582ec856f4fd](https://github.com/JGreenlee/e-mission-phone/commit/24c5856bd3066bf6e9ac367caabd582ec856f4fd)

<table>
<tr>
<td>
On `master`, multiple stub trips
</td>
<td>
On `label_dashboard_profile_sept_2023`, only 1 stub trip
<br>
(remains because it is longer than 5 minutes)
</td>
</tr>
<tr>
<td>

<img src=https://github.com/e-mission/e-mission-docs/assets/15843932/8ceacdda-8bbd-4c9d-b036-d9002b0ee276 width=700>

</td>
<td>

<img src=https://github.com/e-mission/e-mission-docs/assets/15843932/7ca12b1f-75ae-4d3b-99d7-c5935e108a3c width=500>

</td>
</tr>
</table>
